lib/rucio/tests/test_dataset_replicas.py

```diff
@@ -277,14 +279,28 @@ def test_update_collection_replica(self):
         assert dataset_replica['available_replicas_cnt'] == len(files)
         assert dataset_replica['state'] == ReplicaState.AVAILABLE
 
+        # Old behaviour, open empty datasets are not deleted
         # Delete all file replicas -> dataset replica should be deleted
+        cconfig_set(section='reaper', option='remove_open_did', value=False, session=self.db_session)
+        remove_empty_open_did(force_refresh=True, session=self.db_session)
         delete_replicas(rse_id=self.rse_id, files=files, session=self.db_session)
-        update_request = self.db_session.query(models.UpdatedCollectionReplica).filter_by(rse_id=self.rse_id, scope=self.scope, name=dataset_name).one()  # pylint: disable=no-member
-        update_collection_replica(update_request=update_request.to_dict(), session=self.db_session)
-        dataset_replica = self.db_session.query(models.CollectionReplica).filter_by(scope=self.scope, name=dataset_name).all()  # pylint: disable=no-member
-        assert len(dataset_replica) == 0
+        with pytest.raises(NoResultFound):
+            update_collection_replica(update_request=update_request.to_dict(), session=self.db_session)
+
+        # New behaviour, open empty datasets are deleted
+        cconfig_set(section='reaper', option='remove_open_did', value=True, session=self.db_session)
```

Comment:
The config option will persist in the database after the test. Did you try using the config table override as in this example : https://github.com/rucio/rucio/blob/f38bfe391f0c07e023408de6a826c9affba1d5af/lib/rucio/tests/test_conveyor_submitter.py#L96 ?  Config_set will then go a temporary table. Don't foget to also override the cache as here https://github.com/rucio/rucio/blob/f38bfe391f0c07e023408de6a826c9affba1d5af/lib/rucio/tests/test_conveyor_submitter.py#L101
otherwise the cached value can pollute the tests.